test/module/irohad/ametsuchi/postgres_query_executor_test.cpp

```diff
@@ -977,18 +985,478 @@ namespace iroha {
+          const boost::optional<std::string> &req_writer,
+          const boost::optional<std::string> &req_key,
+          const boost::optional<std::string> &first_record_writer,
+          const boost::optional<std::string> &first_record_key,
+          size_t page_size) {
+        auto optional_match = [](const auto &opt, const auto &val) {
+          return not opt or opt.value() == val;
+        };
+
+        Response expected_response;
+        size_t expected_page_size = 0;
+        bool page_started = false;
+        bool page_ended = false;
+        for (const auto &added_writer_and_data : this->added_data_) {
+          const auto &writer = added_writer_and_data.first;
+          const auto &added_data_by_writer = added_writer_and_data.second;
+
+          // check if writer matches query
+          if (optional_match(req_writer, writer)) {
+            for (const auto &key_and_value : added_data_by_writer) {
+              const auto &key = key_and_value.first;
+              const auto &val = key_and_value.second;
+
+              // check if key matches query
+              if (optional_match(req_key, key)) {
+                ++expected_response.total_number;
+                page_started = page_started
+                    or (optional_match(first_record_writer, writer)
+                        and optional_match(first_record_key, key));
+                if (page_started) {
+                  if (page_ended) {
+                    if (not expected_response.next_record) {
+                      expected_response.next_record = RecordId{writer, key};
+                    }
+                  } else {
+                    expected_response.details[writer][key] = val;
+                    ++expected_page_size;
+                    page_ended |= expected_page_size >= page_size;
+                  }
+                }
+              }
+            }
+          }
+        }
+        return expected_response;
+      }
+
+      /**
+       * Compare actual response to the reference one.
+       */
+      void validatePageResponse(
+          const shared_model::interface::AccountDetailResponse &response,
+          const Response &expected_response) {
+        EXPECT_EQ(response.totalNumber(), expected_response.total_number);
+        if (expected_response.next_record) {
+          if (not response.nextRecordId()) {
+            ADD_FAILURE() << "nextRecordId not set!";
+          } else {
+            EXPECT_EQ(response.nextRecordId()->writer(),
+                      expected_response.next_record->writer);
+            EXPECT_EQ(response.nextRecordId()->key(),
+                      expected_response.next_record->key);
+          }
+        } else {
+          EXPECT_FALSE(response.nextRecordId());
+        }
+      }
+
+      /**
+       * Check JSON data of paged response.
+       */
+      void checkJsonData(
+          const std::string &test_data,
+          const std::map<std::string, std::map<std::string, std::string>>
```

Comment:
Maybe make a `using` in fixture?

Reply:
nice, but the best name I could think of was `DetailsByKeyByWriter` :fearful: